boards: raspberrypi: rpi_pico: Use sysbuild.overlay

mcjelcom opened the pull request:
Use `boards/raspberrypi/rpi_pico/sysbuild.overlay` for rpi_pcio to build
SysBuild images.

soburi reviewed (commented):
https://github.com/zephyrproject-rtos/zephyr/pull/90758#discussion_r2113290627

https://github.com/zephyrproject-rtos/zephyr/pull/90758#discussion_r2113290627

There are no other issues besides this one from me.

I will approve it as soon as it is fixed.

jlbwoods commented:
What's the next step to get this merged?

soburi commented:
@nordicjm 
Can you take a look again please?

jlbwoods commented:
> @jlbwoods
> 
> I don't think no need to make sysbuild as default, is there a reason for the change?

True.  Will revert the changes and use storage partitions as the default.

jlbwoods commented:
> @jlbwoods
> 
> I don't think no need to make sysbuild as default, is there a reason for the change?

The reason to make `sysbuild` as default is because most of the overlays inside `mcuboot/boot/zephyr/boards` are using `sysbuild` partition layouts.  The `sysbuild` partitions are "default" with samples like `zephyr/samples/sysbuild/with_mcuboot/` since it is assuming `sysbuild` type partitions and thus there is no `boards` folder in the sample.

soburi commented:
> > @jlbwoods
> > I don't think no need to make sysbuild as default, is there a reason for the change?
> 
> The reason to make `sysbuild` as default is because most of the overlays inside `mcuboot/boot/zephyr/boards` are using `sysbuild` partition layouts. The `sysbuild` partitions are "default" with samples like `zephyr/samples/sysbuild/with_mcuboot/` since it is assuming `sysbuild` type partitions and thus there is no `boards` folder in the sample.

Currently, sysbuild is not the default for any SoC other than those that require different images for two CPUs, such as the ESP32 and NRF91. We do not believe that it is necessary to apply sysbuild as the default for the RP2350, as with the majority of other SoCs.

> > @jlbwoods
> > I don't think no need to make sysbuild as default, is there a reason for the change?
> 
> The reason to make `sysbuild` as default is because most of the overlays inside `mcuboot/boot/zephyr/boards` are using `sysbuild` partition layouts. The `sysbuild` partitions are "default" with samples like `zephyr/samples/sysbuild/with_mcuboot/` since it is assuming `sysbuild` type partitions and thus there is no `boards` folder in the sample.

Currently, sysbuild is not the default for SoCs other than those that require different images for two CPUs, such as the ESP32 and NRF91.
For passing tests, we only require adding overlay files.
Therefore, I believe this change requires a justification based on the use case.

The sysbuild configuration is meaningful if enabled mcuboot, but the RP2350 does not have a reason to make it the default.
(The default must keep simple.)
Without it, the change only reduces the user's usable storage.

I don't believe it is necessary to apply sysbuild as the default for the RP2350, as is the case with most other SoCs.

Commit:
boards: raspberrypi: rpi_pico: Use sysbuild.overlay

Use `boards/raspberrypi/rpi_pico/sysbuild.overlay` for rpi_pcio to build
SysBuild images.

Signed-off-by: John Lin <john.lin@beechwoods.com>

jlbwoods commented:
`sysbuild.overlay` is not yet implemented.
